Source/santactl/Commands/sync/SNTCommandSyncRuleDownload.m

```diff
@@ -124,6 +155,44 @@ - (SNTRule *)ruleFromDictionary:(NSDictionary *)dict {
     newRule.customMsg = customMsg;
   }
 
+  // Check rule for extra notification related info.
+  //
+  // Q: Is the count value sent down with the initial FCM message? Or in each rule?
+  // A: In each rule.
+  //
+  // Q: What happens if the rules get sent before the corresponding FCM message?
+  // A: No notifier object is ever created; user won't receive notification.
+  if (newRule.state == SNTRuleStateWhitelist) {
+    // primaryHash is the bundle hash if there was a bundle hash included in the rule, otherwise
+    // it is simply the binary hash.
+    NSString *primaryHash = dict[kFileBundleHash];
+    if (primaryHash.length != 64) {
+      primaryHash = newRule.shasum;
+    }
+
+    // If we have already seen a rule with the same primary hash, then decrement the count of the
+    // corresponding pending notification.  Otherwise, if this is the first time we've seen this
+    // primary hash, add a count field to the pending notfication and set its initial value.
+    // If the downloaded rule included count information, this initial value is (count - 1).
```

Comment:
this sentence kinda reiterates the code. you might include a point about _why_ you set it to `(count-1)`.